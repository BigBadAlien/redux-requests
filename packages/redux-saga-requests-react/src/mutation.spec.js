import renderer from 'react-test-renderer';
import React from 'react';

import Mutation from './mutation';

describe('Mutation', () => {
  it('renders loading and error', () => {
    const component = renderer.create(
      <Mutation mutation={{ pending: 1, error: 'error' }}>
        {({ loading, error }) => (
          <div>
            {loading && 'loading'}
            {error}
          </div>
        )}
      </Mutation>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders loading and error of mutation with requestKey', () => {
    const component = renderer.create(
      <Mutation requestKey="x" mutation={{ x: { pending: 1, error: 'error' } }}>
        {({ loading, error }) => (
          <div>
            {loading && 'loading'}
            {error}
          </div>
        )}
      </Mutation>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('doesnt crush if an mutation with a requestKey doesnt exist', () => {
    const component = renderer.create(
      <Mutation requestKey="y" mutation={{ x: { pending: 1, error: 'error' } }}>
        {({ loading, error }) => (
          <div>
            {loading && 'loading'}
            {error}
          </div>
        )}
      </Mutation>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders custom component as prop with extra prop', () => {
    const MutationComponent = ({ loading, error, extra }) => (
      <div>
        {loading && 'loading'} {error} {extra}
      </div>
    );

    const component = renderer.create(
      <Mutation
        mutation={{ pending: 1, error: 'error' }}
        extra="extra"
        component={MutationComponent}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
