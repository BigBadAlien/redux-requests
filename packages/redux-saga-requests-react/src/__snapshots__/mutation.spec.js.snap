// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation doesnt crush if an mutation with a requestKey doesnt exist 1`] = `<div />`;

exports[`Mutation renders custom component as prop with extra prop 1`] = `
<div>
  loading
   
  error
   
  extra
</div>
`;

exports[`Mutation renders loading and error 1`] = `
<div>
  loading
  error
</div>
`;

exports[`Mutation renders loading and error of mutation with requestKey 1`] = `
<div>
  loading
  error
</div>
`;
